lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -0,0 +1,474 @@
+import pro.taskana.TaskanaConfiguration.Builder;
+import pro.taskana.common.api.TaskanaEngine;
+import pro.taskana.common.internal.util.IdGenerator;
+import pro.taskana.common.internal.util.Pair;
+import pro.taskana.simplehistory.impl.SimpleHistoryServiceImpl;
+import pro.taskana.simplehistory.impl.jobs.HistoryCleanupJob;
+import pro.taskana.spi.history.api.TaskanaHistory;
+import pro.taskana.spi.history.api.events.task.TaskHistoryEvent;
+import pro.taskana.spi.history.api.events.task.TaskHistoryEventType;
+import pro.taskana.testapi.TaskanaConfigurationModifier;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.WithServiceProvider;
+import pro.taskana.testapi.security.WithAccessId;
+
+@WithServiceProvider(
+    serviceProviderInterface = TaskanaHistory.class,
+    serviceProviders = SimpleHistoryServiceImpl.class)
+@TaskanaIntegrationTest
+class HistoryCleanupJobAccTest {
+
+  @TaskanaInject TaskanaEngine taskanaEngine;
+
+  SimpleHistoryServiceImpl historyService;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    historyService = new SimpleHistoryServiceImpl();
+    historyService.initialize(taskanaEngine);
+  }
+
+  @WithAccessId(user = "admin")
+  @BeforeEach
+  void before() throws Exception {
+    // clean previously created task histories
+    List<TaskHistoryEvent> taskHistoryEventList = historyService.createTaskHistoryQuery().list();
+    List<String> taskIds = taskHistoryEventList.stream().map(TaskHistoryEvent::getTaskId).toList();
+    if (!taskIds.isEmpty()) {
+      historyService.deleteHistoryEventsByTaskIds(taskIds);
+    }
+  }
+
+  @Test
+  @WithAccessId(user = "admin")
+  void should_NotCleanHistoryEventsUntilDate_When_MinimumAgeNotReached() throws Exception {
+
+    TaskHistoryEvent eventToBeCleaned =
+        createTaskHistoryEvent(
+            "wbKey1",
+            "taskId1",
+            TaskHistoryEventType.CREATED.getName(),
+            "wbKey2",
+            "someUserId",
+            "someDetails");
+    eventToBeCleaned.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+    eventToBeCleaned.setParentBusinessProcessId("sameParentId");
+
+    TaskHistoryEvent eventToBeCleaned2 =
+        createTaskHistoryEvent(
+            "wbKey1",
+            "taskId1",
+            TaskHistoryEventType.COMPLETED.getName(),
+            "wbKey2",
+            "someUserId",
+            "someDetails");
+    eventToBeCleaned2.setCreated(Instant.now().minus(5, ChronoUnit.DAYS));
+    eventToBeCleaned2.setParentBusinessProcessId("sameParentId");
+
+    historyService.create(eventToBeCleaned);
+    historyService.create(eventToBeCleaned2);
+
+    HistoryCleanupJob job = new HistoryCleanupJob(taskanaEngine, null, null);
+    job.run();
+
+    assertThat(historyService.createTaskHistoryQuery().count()).isEqualTo(2);
+  }
+
+  private TaskHistoryEvent createTaskHistoryEvent(
```

Comment:
What do you think about extending this method by created and parentBusinessId?

Reply:
Good idea, I have extended it and adjusted the tests